File: LeetCode-75/746-E-Min-Cost-Climbing-Stairs/min-cost-climb-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: list[int]) -> int:
        prev1, prev2, length = 0, 0, len(cost)
        for i in range(2, length + 1):
            prev2, prev1 = prev1, min(cost[i-1] + prev1, cost[i-2] + prev2)
        return prev1

File: LeetCode-75/746-E-Min-Cost-Climbing-Stairs/min-cost-climb-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int prev1=0;
        int prev2=0;
        int curr = 0;
        int len = cost.length;
        for (int k=2; k<=len; k++)
        {
            curr = Math.min(cost[k-1]+prev1, cost[k-2]+prev2);
            prev2=prev1;
            prev1=curr;
        }
        return prev1;
    }
}